autoload/indent_guides.vim

```diff
@@ -278,9 +277,26 @@ endfunction
 "
 function! indent_guides#exclude_filetype()
   for ft in split(&ft, '\.')
-    if index(g:indent_guides_exclude_filetypes, ft) > -1
+    if index(indent_guides#getvar('indent_guides_exclude_filetypes'), ft) > -1
       return 1
     end
   endfor
   return 0
 endfunction
+
+"
+" Return variables value
+" Choose local buffer variable first if exist or global variable if not
+" return -1 if none of local buffer / global variable exists
+"
+function! indent_guides#getvar(var)
+  let varName=a:var
+  if (exists ("b:" . varName))
+    exe "let retVal=b:" . varName
+  elseif (exists ("g:" . varName))
+    exe "let retVal=g:" . varName
+  else
+    exe "let retVal=-1"
+  endif
+  return retVal
+endfunction
```

Comment:
Would it work to simplify this function and also avoid the repetitive "indent_guides" in both the function name and var name and only need to use it in the function name?

```suggestion
function! indent_guides#getvar(var)
  let l:varName = "indent_guides_" . a:var
  return get(b:, l:varName, get(g:, l:varName, -1))
endfunction
```

Reply:
Applied